opencog/atoms/bind/PatternLink.cc

```diff
@@ -191,6 +194,11 @@ PatternLink::PatternLink(const std::set<Handle>& vars,
 			_pat.mandatory.push_back(h);
 		}
 	}
+
+	// If there are any disconnected optionals, consider each of them as an
+	// optional for all mandatory clauses in the pattern
+	_pat.optionals.insert(discon_opts.begin(), discon_opts.end());
```

Comment:
It was meant to insert only disconnected optionals to the `_pat` but not those that are connected to some other patterns, so just doing `_pat.optionals = opts;` may not be a good idea here. But now thinking about it again, it seems this is not the right place to put `discon_opts`, let me correct this.